app/models/pawn.rb

```diff
@@ -11,6 +11,16 @@ def valid_move?(x_end, y_end)
 
   private
 
+  def diagonal_capture?(x_end, y_end)
+    if color == 'white'
+      return true if game.pieces.where(horizontal_position: x_end, vertical_position: y_end, color: 'black').present?
+      game.pieces.where(horizontal_position: x_end, vertical_position: vertical_position, color: 'black', passant: true).present?
+    else
+      return true if game.pieces.where(horizontal_position: x_end, vertical_position: y_end, color: 'white').present?
+      game.pieces.where(horizontal_position: x_end, vertical_position: vertical_position, color: 'white', passant: true).present?
+    end
+  end
+
```

Comment:
This can be:

``` ruby
def diagonal_capture?(x_end, y_end)
  opposite_color = color == 'white' ? 'black' : 'white'
  return true if game.pieces.where(horizontal_position: x_end, vertical_position: y_end, color: opposite_color).present?
  game.pieces.where(horizontal_position: x_end, vertical_position: vertical_position, color: opposite_color, passant: true).present?
end
```